tests/src/test/scala/common/rest/WskRest.scala

```diff
@@ -0,0 +1,1434 @@
+   * if the code is anything but DONTCARE_EXIT, assert the code is as expected
+   */
+  override def list(expectedExitCode: Int = OK.intValue, nameSort: Option[Boolean] = None)(
+    implicit wp: WskProps): RestResult = {
+    val entPath = Path(s"$basePath/namespaces")
+    val resp = requestEntity(GET, entPath)
+    val result = if (resp == None) new RestResult(NotFound) else new RestResult(resp.status, getRespData(resp))
+    validateStatusCode(expectedExitCode, result.statusCode.intValue)
+    result
+  }
+
+  /**
+   * Gets entities in namespace.
+   *
+   * @param namespace (optional) if specified must be  fully qualified namespace
+   * @param expectedExitCode (optional) the expected exit code for the command
+   * if the code is anything but DONTCARE_EXIT, assert the code is as expected
+   */
+  override def get(namespace: Option[String] = None,
+                   expectedExitCode: Int = OK.intValue,
+                   nameSort: Option[Boolean] = None)(implicit wp: WskProps): RestResult = {
+    val (ns, _) = namespace map { this.getNamespaceActionName(_) } getOrElse (wp.namespace, "")
+    val entPath = Path(s"$basePath/namespaces/$ns/")
+    val resp = requestEntity(GET, entPath)
+    val r = new RestResult(resp.status, getRespData(resp))
+    validateStatusCode(expectedExitCode, r.statusCode.intValue)
+    r
+  }
+
+  /**
+   * Looks up namespace for whisk props.
+   *
+   * @param wskprops instance of WskProps with an auth key to lookup
+   * @return namespace as string
+   */
+  override def whois()(implicit wskprops: WskProps): String = {
+    val ns = list().getBodyListString
+    if (ns.size > 0) ns(0).toString() else ""
+  }
+}
+
+class WskRestPackage
+    extends RunWskRestCmd
+    with ListOrGetFromCollectionRest
+    with DeleteFromCollectionRest
+    with BasePackage {
+
+  override protected val noun = "packages"
+
+  /**
+   * Creates package. Parameters mirror those available in the REST.
+   *
+   * @param name either a fully qualified name or a simple entity name
+   * @param expectedExitCode (optional) the expected exit code for the command
+   * if the code is anything but DONTCARE_EXIT, assert the code is as expected
+   */
+  override def create(name: String,
+                      parameters: Map[String, JsValue] = Map(),
+                      annotations: Map[String, JsValue] = Map(),
+                      parameterFile: Option[String] = None,
+                      annotationFile: Option[String] = None,
+                      shared: Option[Boolean] = None,
+                      update: Boolean = false,
+                      expectedExitCode: Int = OK.intValue)(implicit wp: WskProps): RestResult = {
+    val path = getNamePath(noun, name)
+    var bodyContent = JsObject("namespace" -> s"${wp.namespace}".toJson, "name" -> name.toJson)
+
+    val (params, annos) = this.getParamsAnnos(parameters, annotations, parameterFile, annotationFile)
+    if (!update) {
+      val published = shared map { s =>
```

Comment:
Here you can do `shared.getOrElse(false)`.

Reply:
done